packages/gatsby/src/redux/actions/public.js

```diff
@@ -414,6 +414,7 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     let pageMode: PageMode = `SSG`
     if (page.defer) {
       pageMode = `DSR`
+      internalPage.defer = true
```

Comment:
Need to propagate `defer` to the page instance. Otherwise `deletePage`/`createPage` pattern often used in `onCreatePage` ignores `defer` flag.